### PR TITLE
Bugfix/nicps 155

### PR DESCRIPTION
**Pull request by deepakzimb**
Rebase snickers to 8.8.8 tag

### CODE REVIEW

**Comment by k-kato** [comment]
LGTM

**Comment by gordyt** [comment]
@deepakzimb Not sure this is correct.  Couple of things...

- looks like you are doing a merge, not a rebase (and a rebase is what you want to to)
- Are you sure that you want to be rebasing on the `8.8.8` tag?  Usually feature work is rebased on top of the `HEAD` of the `develop` branch, not a tag on the `master` branch.

File: store/conf/attrs/zimbra-attrs.xml [hunk 1]
```diff
@@ -9657,8 +9667,5 @@ TODO: delete them permanently from here
   <desc>This attribute is used to enable/disable CAPTCHA functionality on login page</desc>
 </attr>
 
-<attr id="5006" name="zimbraCAPTCHALoginFailedCount" type="integer" cardinality="single" optionalIn="account" flags="domainAdminModifiable" since="8.8.6">
```
[the remark this answers is not in the document]
Comment (deepakzimb) [reply store/conf/attrs/zimbra-attrs.xml]: Added back in new commit.

File: store/conf/attrs/zimbra-attrs.xml [hunk 2]
```diff
@@ -9465,7 +9466,7 @@ TODO: delete them permanently from here
     to this proxy server after which login is rejected with an
     IMAP bye response. This counter is
     cumulative for all users that appear to the proxy to be logging in
-    from the same IP address and IMAP/IMAPS protocol.
```
[the remark this answers is not in the document]
Comment (deepakzimb) [reply store/conf/attrs/zimbra-attrs.xml]: Sorry but I didn't fine white space in VI editor. Could you please share one line so I will confirm again. Thanks.

File: store/src/java/com/zimbra/cs/imap/RemoteImapMailboxStore.java [hunk 3]
```diff
@@ -68,7 +68,27 @@ public RemoteImapMailboxStore(ZMailbox mailbox, String accountId) {
...
+         * spotted a change because the WaitSet last known change will be 0 but we want to be at least
+         * as up to date as the associated ZMailbox.  At this point, our view should be at least as
+         * up to date as ZMailbox, so this shouldn't cause loss of updates.
+         */
+        irs.updateLastChangeId(irs.getMailbox().getLastChangeID());
+        return irs;
+    }
+
+    @Override
+    public List<ImapListener> getListeners(ItemIdentifier ident) {
```
Comment (k-kato) [inline store/src/java/com/zimbra/cs/imap/RemoteImapMailboxStore.java]: The function 

    public List<ImapListener> getListeners(ItemIdentifier ident) {

is defined twice. Build should fails.
Please make sure you can build all components (native, common, soap, client and store) and run "ant test-all" before you send/update the PR.

File: store/conf/attrs/zimbra-attrs.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (k-kato) [reply store/conf/attrs/zimbra-attrs.xml]: Please run "git diff NICPS-155-check bugfix/NICPS-155". You can see a lot of changes in zimbra-attrs.xml

File: store/src/java/com/zimbra/cs/imap/RemoteImapMailboxStore.java (hunk 3 above)
Comment (deepakzimb) [reply store/src/java/com/zimbra/cs/imap/RemoteImapMailboxStore.java]: Fixed.

About this "Please run "git diff NICPS-155-check bugfix/NICPS-155". You can see a lot of changes in zimbra-attrs.xml" --

I saw changes, that are similar diff, same line is removed and added back.

**Pull request merged**